sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -651,13 +652,31 @@ class SparkSession private(
     withActive {
       val plan = tracker.measurePhase(QueryPlanningTracker.PARSING) {
         val parsedPlan = sessionState.sqlParser.parsePlan(sqlText)
-        if (args.nonEmpty) {
-          PosParameterizedQuery(parsedPlan, args.map(lit(_).expr).toImmutableArraySeq)
-        } else {
-          parsedPlan
+        parsedPlan match {
+          case compoundBody: CompoundBody => compoundBody
+          case logicalPlan: LogicalPlan if args.nonEmpty =>
+            PosParameterizedQuery(logicalPlan, args.map(lit(_).expr).toImmutableArraySeq)
+          case p =>
+            assert(args.isEmpty, "Named parameters are not supported for batch queries")
```

Comment:
Is it user visible error? If so, let's create an error condition in `UNSUPPORTED_FEATURE`.

Reply:
Ok, here we should not even assert because `p` can be `LogicalPlan` with no arguments. The question is what if we have `CompoundBody` with arguments, thenwe should maybo throw an error. Do you agree?

Reply:
Done.